src/git/zcl_abapgit_git_porcelain.clas.abap

```diff
@@ -4,31 +4,32 @@ CLASS zcl_abapgit_git_porcelain DEFINITION
   CREATE PUBLIC .
 
   PUBLIC SECTION.
-
-    TYPES:
-      BEGIN OF ty_pull_result,
-        files   TYPE zif_abapgit_definitions=>ty_files_tt,
-        objects TYPE zif_abapgit_definitions=>ty_objects_tt,
-        branch  TYPE zif_abapgit_definitions=>ty_sha1,
-      END OF ty_pull_result .
-    TYPES:
-      BEGIN OF ty_push_result,
-        new_files     TYPE zif_abapgit_definitions=>ty_files_tt,
-        branch        TYPE zif_abapgit_definitions=>ty_sha1,
-        updated_files TYPE zif_abapgit_definitions=>ty_file_signatures_tt,
-        new_objects   TYPE zif_abapgit_definitions=>ty_objects_tt,
-      END OF ty_push_result .
+    TYPES: BEGIN OF ty_pull_result,
+             files   TYPE zif_abapgit_definitions=>ty_files_tt,
+             objects TYPE zif_abapgit_definitions=>ty_objects_tt,
+             branch  TYPE zif_abapgit_definitions=>ty_sha1,
+             commit  TYPE zif_abapgit_definitions=>ty_sha1,
+           END OF ty_pull_result .
+    TYPES: BEGIN OF ty_push_result,
+             new_files     TYPE zif_abapgit_definitions=>ty_files_tt,
+             branch        TYPE zif_abapgit_definitions=>ty_sha1,
+             updated_files TYPE zif_abapgit_definitions=>ty_file_signatures_tt,
+             new_objects   TYPE zif_abapgit_definitions=>ty_objects_tt,
+           END OF ty_push_result .
 
     CLASS-METHODS pull
       IMPORTING
+        !io_repo         TYPE REF TO zcl_abapgit_repo
```

Comment:
I'd like to keep the repo out of the git package (/src/git/), done multiple refactorings to get to the current state